server/routes/induction/update/highestLevelOfEducationUpdateController.test.ts

```diff
@@ -0,0 +1,407 @@
+    it('should get the Highest Level of Education view given there is no HighestLevelOfEducationForm on the session', async () => {
+      // Given
+      const prisonNumber = 'A1234BC'
+      req.params.prisonNumber = prisonNumber
+
+      const prisonerSummary = aValidPrisonerSummary()
+      req.session.prisonerSummary = prisonerSummary
+      const inductionDto = aLongQuestionSetInductionDto()
+      req.session.inductionDto = inductionDto
+      req.session.highestLevelOfEducationForm = undefined
+
+      const expectedHighestLevelOfEducationForm = {
+        educationLevel: EducationLevelValue.SECONDARY_SCHOOL_TOOK_EXAMS,
+      }
+
+      const expectedView = {
+        prisonerSummary,
+        form: expectedHighestLevelOfEducationForm,
+        backLinkUrl: '/plan/A1234BC/view/education-and-training',
+        backLinkAriaText: 'Back to <TODO - check what CIAG UI does here>',
+        errors,
+      }
+
+      // When
+      await controller.getHighestLevelOfEducationView(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(res.render).toHaveBeenCalledWith(
+        'pages/induction/prePrisonEducation/highestLevelOfEducation',
+        expectedView,
+      )
+      expect(req.session.highestLevelOfEducationForm).toBeUndefined()
+      expect(req.session.inductionDto).toEqual(inductionDto)
+    })
+
+    it('should get the Highest Level of Education view given there is a HighestLevelOfEducationForm already on the session', async () => {
+      // Given
+      const prisonNumber = 'A1234BC'
+      req.params.prisonNumber = prisonNumber
+
+      const prisonerSummary = aValidPrisonerSummary()
+      req.session.prisonerSummary = prisonerSummary
+      const inductionDto = aLongQuestionSetInductionDto()
+      req.session.inductionDto = inductionDto
+
+      const expectedHighestLevelOfEducationForm = {
+        educationLevel: EducationLevelValue.SECONDARY_SCHOOL_TOOK_EXAMS,
+      }
+      req.session.highestLevelOfEducationForm = expectedHighestLevelOfEducationForm
+
+      const expectedView = {
+        prisonerSummary,
+        form: expectedHighestLevelOfEducationForm,
+        backLinkUrl: '/plan/A1234BC/view/education-and-training',
+        backLinkAriaText: 'Back to <TODO - check what CIAG UI does here>',
+        errors,
+      }
+
+      // When
+      await controller.getHighestLevelOfEducationView(
+        req as undefined as Request,
+        res as undefined as Response,
+        next as undefined as NextFunction,
+      )
+
+      // Then
+      expect(res.render).toHaveBeenCalledWith(
+        'pages/induction/prePrisonEducation/highestLevelOfEducation',
+        expectedView,
+      )
+      expect(req.session.highestLevelOfEducationForm).toBeUndefined()
+      expect(req.session.inductionDto).toEqual(inductionDto)
+    })
+  })
+
+  describe('submitHighestLevelOfEducationForm', () => {
```

Comment:
The scenarios for submitting the Highest Level of Education form that have been implemented and therefore unit tested in this PR are:

* submit form with validation errors
* submit form with no changes to highest level of education
* Induction already contains previous qualifications (and therefore a highest level of education), change the highest level of education to one that does not require qualifications (ie. left school before exams). Assert that the induction is updated and the existing qualifications are kept
* Induction already contains previous qualifications (and therefore a highest level of education), change the highest level of education to one that does require qualifications (ie. took exams). Assert that the induction is updated and the existing qualifications are kept
* Induction contains no previous qualifications, change the highest level of education to one that does not require qualifications (ie. left school before exams). Assert that the induction is updated and the existing qualifications (empty collection) are kept
* submit form, error calling induction service


The scenario that has not been implemented or tested yet is that where the Induction contains no previous qualifications and the change in Highest Level of Education is one where the prisoner took exams after school and therefore requires that the user is taken through a mini-flow to enter the qualification details.